File: app/src/main/java/com/ardev/builder/project/Project.java
package com.ardev.builder.project;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.ardev.builder.project.api.Module;
import com.ardev.builder.project.impl.AndroidModuleImpl;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Set;
import java.util.TreeMap;


public class Project {
 	private final Module mMainModule;
    private final File mRoot;

    

    private volatile boolean mCompiling;

  
    
    public Project(File root) {
        mRoot = root;
      mMainModule = new AndroidModuleImpl(new File(mRoot, "app"));
       
    }

    public boolean isCompiling() {
        return mCompiling;
    }

    public void setCompiling(boolean compiling) {
        mCompiling = compiling;
    }

   

    @NonNull
    public Module getMainModule() {
        return mMainModule;
    }
    
	public void addModule( File library) {
		
		}

    public File getRootFile() {
        return mRoot;
    }
	
	public void open() throws IOException {
        mMainModule.open();
	}
	
	public void index() throws IOException {
        
            mMainModule.clear();
            mMainModule.index();
        }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Project project = (Project) o;
        return mRoot.equals(project.mRoot);
    }

    @Override
    public int hashCode() {
        return Objects.hash(mRoot);
    }


}

File: app/src/main/java/com/ardev/builder/project/Project.kt
package com.ardev.builder.project

import java.io.File
import java.io.IOException
import com.ardev.builder.project.api.Module
import com.ardev.builder.project.impl.AndroidModuleImpl

class Project(private val root: File) {
      
    val rootFile: File = root
      
    val name: String = rootFile.name
      
    val mainModule: Module lazy by { createMainModule(rootFile) }
    
    @Volatile            
    var isCompiling: Boolean = false
       
    fun addModule(library: File) {
       
    }

    

  
    fun open() {
        mainModule.open()
    }

   
    fun index() {
        mainModule.clear()
        mainModule.index()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val project = o as Project
        return root == project.root
    }

    override fun hashCode(): Int {
        return root.hashCode()
    }
}
